build.gradle

```diff
@@ -71,6 +72,52 @@ subprojects {
     apply plugin: 'idea'
     apply plugin: 'eclipse'
     apply plugin: 'java'
+    apply plugin: 'com.chrisgahlert.gradle-dcompose-plugin'
+
+    // Using gradle-dcompose-plugin to build/push docker images
+    dcompose {
+        if (rootProject.hasProperty('dockerRegistry')) {
+            def registryName = rootProject.getProperty('dockerRegistry')
+            def isDockerHub = rootProject.hasProperty('useDockerHub') ? rootProject.getProperty('useDockerHub') : false
+
+            def repositoryName = "${project.name - 'Source' + '-source'}".toLowerCase()
+            if (rootProject.hasProperty('repositoryName')) {
+                repositoryName = rootProject.getProperty('repositoryName')
+            }
+
+            connector {
+                buildFiles = project.copySpec {
+                    from "$projectDir/src/main/docker/"
+                    from "$projectDir/build/distributions/${project.name}.tar"
+                    if (rootProject.hasProperty('connectorSpecificInclusions')) {
+                        def inclusionsDir = rootProject.getProperty('connectorSpecificInclusions')
+                        from "$inclusionsDir"
+                    }
+                }
+
+                repository = registryName + repositoryName
+                if (rootProject.hasProperty('imageTag')) {
+                    repository = registryName + repositoryName + rootProject.getProperty('imageTag')
+                }
+                logBuildStatus = true
+                buildLogFile = file("$buildDir/dockerBuildLog.txt")
+            }
+
+            // The credentials here should be stored in gradle.properties.
+            if (isDockerHub) {
+                dockerClientConfig = {
+                    withRegistryUsername dockerRegistryUser
+                    withRegistryPassword dockerRegistryPassword
+                }
+            } else {
+                registry(registryName) {
```

Comment:
Here, registry name is something like `quay.io/fred` or `/fred`, where the last one means to default to Docker Hub.  Correct?